python/lsst/daf/butler/core/storedFileInfo.py

```diff
@@ -45,8 +47,9 @@ class StoredFileInfo(StoredDatastoreItemInfo):
     __slots__ = {"formatter", "path", "storageClass", "component",
                  "checksum", "file_size"}
 
-    formatter: str
-    """Fully-qualified name of Formatter."""
+    formatter: Union[str, Formatter, Type[Formatter]]
```

Comment:
This is quite problematic since it stops us from saying that we accept these three types but it will always be stored internally as a str.

Reply:
I think we had a reason for using `__post_init__` since it seemed designed for this and didn't result in me trying to ensure that I implemented `__init__` for the dataclass properly.

Reply:
I've rewritten it and things seem to work okay now.